### DISCUSSION TITLE
Peak Shape / Beam Shape visualization

### DISCUSSION THREAD

**Discussion by noahmclean** [CIRDLES/Tripoli#44] [Show and tell]
Tripoli 0.0.4-alpha has a demonstration tool to deconvolve peak shapes measured during peak centering (on Isotopx Phoenix for now) into the shape and width of the ion beam.  This discussion thread is for developing this visualization tool -- how should this information be presented, visualized, organized, quantified, and saved for later?

**Comment by ryanickert** [comment]
For quantification:

If the deconvolved peak shape is roughly gaussian, then my primary request would be to calculate variance (~beam width) skewness (beam symmetry) and kurtosis (peak broadening? tails? I'm not sure what this means but it would be great to have a few thousand of them to figure out).  A flag for bimodality would be good as well.  On the peak shape itself, calculating resolution would be useful, and also more caveman metrics like peak flat and shoulder width in apparent amu.

Secondary request would be undertainties on these estimates - in the case of ion counter measurements, derived from counting stats.  This would make it easy to compare a peak shapes from small and big beams.

Tertiary request would be to do this and account for beam drift.

**Reply by noahmclean** [reply]
For beam width, my idea was to going to provide the width at 1% (or 2%? 5%?) of max intensity, in both amu and mm.  E.g., width should be about 0.35 mm on a standard 0.3 mm slit and second order focusing.  The moments of the ion beam distribution sound good, though the units can be hard to interpret.  I like the idea of measuring resolution, peak flat, and shoulder width.  

Uncertainties will be harder but are possible.  From my experiments, it looks like even small beams (a few 1000 cps) still produce beam widths good to 0.02 mm or so... enough to tell if the focus is way off, though the detailed shape may not be resolvable.  

One big question is how do you want to see and interact with this information?  After a run, Tripoli would just bring in a folder of these... perhaps several peak scans per block over several blocks.  Scans could be different isotopes on the same collector or on different collectors.  There will be graphical data (peak scans and fits, beam shapes) and numerical data.

**Reply by ryanickert** [reply]
I hear what you're saying about units, but it would be great to see some metric of (a)symmetry over time, so skew would be great.  Seems like a natural choice if you're already inverting for beam shape, but a metric more closely related to the peak shape would work too.

Most of these metrics are not particularly meaningful in isolation, so I would like the ability to display them as a time series, with a way to easily zoom in and out to look at either barrel-to-barrel changes (over weeks/months) or within barrel drifts.  I'm picturing a series of stacked OG-Tripoli-like panels with shoulder width, flat width, and skew.  I'm not sure how best to deal with data clumping, since you're getting as many as 200 Pb scans per barrel, but my initial instinct (without seeing what it looks like) is to plot 'em all.

**Comment by bowring** [comment]
It would be great to get a visual mockup of these ideas to help drive implementation.  Drawings are good first passes.

**Reply by ryanickert** [reply]
![image](https://user-images.githubusercontent.com/7709328/181798476-34935c08-c0e5-4f30-be7e-9bd9dbeb053f.png)

But with absolute time on the x axis and these peak shape parameters on the y axis.

**Comment by bowring** [comment]
per discussion with @noahmclean 

plot#1 peak width = distance between two red dots on beam shape plot vs time for the contents of the PeakCentres folder - Noah will supply exemplar folder - one plot for each peak present.  Each set of mass plus detector will be on its own graph.  Users will be able to click on points and pop up images of the beam shape and the peak shape.
